### ISSUE TITLE
Escaped double-quote in Objective-C string confuses parser

### ISSUE THREAD

**Issue by smokris** [doxygen/doxygen#9771]
**Describe the bug**
When I escape a double-quote inside an Objective-C string (e.g. `@"\""`), Doxygen 1.9.6's parser gets confused and ignores the rest of the file.

**Expected behavior**
Doxygen should be able to parse `@"\""` in an Objective-C source file.

**To Reproduce**
```console
$ cat > test.m
/**
 * …documentation for SomeClass…
 */
@interface SomeClass : NSObject
- (void)foo;
@end

@implementation SomeClass
- (NSString *)description
{
	return @"\"";
}

/**
 * …documentation for foo…
 */
- (void)foo
{
}
@end

^D

$ doxygen -g
$ doxygen
warning: Member foo() (function) of class SomeClass is not documented.
```

**Version**
The unexpected warning occurs when using Doxygen 1.9.6 (built by Homebrew) on macOS 13.1.

**Additional context***
With Doxygen 1.9.5, this example worked correctly (Doxygen correctly found the documentation for the `-foo` method and emitted HTML containing that method's documentation), but Doxygen 1.9.6 says that method is not documented.

**Comment by albert-github** [comment]
I tried your example and with the 1.9.6 and 1.9.5 versions I got the warning as indicated. Furthermore I didn't see in either version the documentation of `foo`.
- are you sure you didn't see the warning with 1.9.5 or is the example a but to simplistic compared to the case you used for the 1.9.5 version?
- I still think this is a bug

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #9774

**Comment by smokris** [comment]
Thanks!  Yes, it looks like I simplified the example a little too much.  Here's a revised snippet that works as expected on Doxygen 1.9.5 (and the current `master` branch 👍🏼), but emits a warning on 1.9.6:
```objc
/**
 * …documentation for SomeClass…
 */
@interface SomeClass : NSObject
- (void)foo;
@end

@implementation SomeClass
- (NSString *)description
{
    return [NSString stringWithFormat:@"SomeClass(\"%@\")", someInstanceVariable];
}

/**
 * …documentation for foo…
 */
- (void)foo
{
}
@end
```

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.7.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).

**Comment by smokris** [comment]
> Please verify if this is indeed the case.

Thanks.  Confirmed fixed in Doxygen 1.9.7.
